### PR TITLE
Upgrade Dependencies

### PR DESCRIPTION
**Pull request by xeptore**
Upgrade all dependencies to their latest version.

Closes martinmcwhorter/commitiquette#4.

### CODE REVIEW

**Comment by xeptore** [comment]
@martinmcwhorter:

- [x] Pass all the tests
- [x] Upgrade all external dependencies to their latest version
- [x] Remove any unneeded dependencies
- [x] Make linters happy
- [x] More strict checks (e.g., using `===` instead of `==`)

**Comment by xeptore** [comment]
@martinmcwhorter, the only part that I'm not sure whether it is migrated correctly or not is the [`husky`](https://github.com/typicode/husky/) `git` commit hooks, as mentioned in the following snippet:
https://github.com/xeptore/commitiquette/blob/a77e01b29133843ccfd3bf410b9fa41357700009/package.json#L61-L63

I configured husky to initialize itself in the [`npm` `prepare` script](https://github.com/xeptore/commitiquette/pull/1/commits/a77e01b29133843ccfd3bf410b9fa41357700009) as described in its [`v6.0.0` release notes](https://github.com/typicode/husky/releases/tag/v6.0.0) and as I noticed, its `pre-commit` hook, which executes all the test suites is working fine, but for the other hooks, `prepare-commit-msg`, and `commit-msg` hooks, I could not realize what is going on, to be honest.

So, I'll leave this to-do task as undone for now:
- [ ] Ensure husky hooks installation

**Comment by xeptore** [comment]
OK, finally [I migrated husky and its configurations to its latest version](https://github.com/xeptore/commitiquette/pull/1/commits/b9b0af0dd5820acb9de416709f2e561fe3acc5dc) correctly using the following official documentations and tools:

- https://typicode.github.io/husky/#/?id=husky-4-to-6-cli
- https://github.com/typicode/husky-4-to-6

I also updated hook scripts manually in order to be executed correctly (mostly by `npx`).

I mark the respective to-do task as done.

**Comment by xeptore** [comment]
I think this pull request is ready to be merged into the main repository. I tried to introduce minimum changes in the internal working mechanisms of the application.

Feel free to make corrections 😃

File: src/filters.ts [hunk 1]
```diff
@@ -24,14 +25,14 @@ export function leadingBlankFilter(value: string, rule: Rule<undefined> | undefi
   return value.trimLeft();
 }
 
-export function fullStopFilter(value: string, rule: Rule<string> | undefined): string {
-  if (rule == null) {
+export function fullStopFilter(value: string, rule: QualifiedRules['subject-full-stop'] | undefined): string {
+  if (rule === undefined) {
```
Comment (martinmcwhorter) [inline src/filters.ts:29]: This is one of those counter intuitive JavaScript things. 

(rule == null) is actually safer than (rule === undefined) 

(rule == null) is equivalent to (rule === undefined || rule === null)

Comment (xeptore) [reply src/filters.ts:29]: Yes, you are right about `== null` is safer than `=== undefined`, but if you see the type declaration of rules in the `@commitlint/types` package, they declared missing rule or their missing values as `undefined`, while they could specify it as `null` if they were not sure. I think they will handle the missing values and missing rule configurations correctly before returning the user configuration from `load` function of `@commitlint/load` package, and they should handle all the possibilities and missing values beforehand, as the type declaration says.

Comment (xeptore) [reply src/filters.ts:29]: In general, I think there are cases in this application that are not, more precisely, _should not be_ our concerns, and checking **all** the cases in our code will make it much harder in the future to change. Moreover, as I realized, we are not directly facing the user configurations, and we receive what `@commitlint/load` loads (and maybe validates, parses, or even converts before returning the configuration to our code), thus, we should rely on what we get and see, which is in this case is the type declarations that _they_ have specified.

File: src/engine.ts [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
-function buildQuestions(rules: Rules) {
+function buildQuestions(rules: QualifiedRules) {
   const combinedQuestions = pipeWith<Question[]>(
     [],
-    x => typeMaker(x, rules),
-    x => scopeMaker(x, rules),
-    x => subjectMaker(x, rules),
-    x => bodyMaker(x, rules),
-    x => footerMaker(x, rules)
+    (x) => typeMaker(x, rules),
```
Comment (martinmcwhorter) [inline src/engine.ts:16]: I am a big fan of letting prettier do its thing, but for this lets add the rule ` "arrowParens": "avoid"` to the .prettierrc

Comment (xeptore) [reply src/engine.ts:16]: Yes, I agree. I will change it in an upcoming commit.

File: src/filters.ts (hunk 1 above)
Comment (martinmcwhorter) [reply src/filters.ts:29]: I agree 100% whole heartedly that we should trust and rely on comimit and commitizen. 

The argument type is still `RuleConfigTuple<undefined> | undefined` so this won't affect what we can use for that argument. 

Let's leave the (rule == null) for now, just to be safe -- reduce the changes in the PR. 

If in the future this causes issues, we can change it in isolation.

Comment (xeptore) [reply src/filters.ts]: OK, reverted at: https://github.com/xeptore/commitiquette/pull/1/commits/26a9e8b136cdc900709dda9c3a8b268dc5b231c3.

**Review by martinmcwhorter — APPROVED** [review]

**Pull request merged**